pkg/common/cns-lib/volume/taskmanager.go

```diff
@@ -0,0 +1,355 @@
+	t.taskToChannelMap[TaskDetails{Reference: taskMoRef, MarkedForRemoval: true}] = ch
+	return nil
+}
+
+func (t *TaskViewManagerImpl) GetPendingTasksCount() int {
+	t.mu.Lock()
+	defer t.mu.Unlock()
+	return len(t.taskToChannelMap)
+}
+
+func NewTaskViewManager(virtualCenter *cnsvsphere.VirtualCenter) *TaskViewManagerImpl {
+	ctx := context.Background()
+	m := make(map[TaskDetails]chan Result)
+	t := &TaskViewManagerImpl{
+		virtualCenter:    virtualCenter,
+		ctx:              ctx,
+		taskToChannelMap: m,
+	}
+	go t.init(t.ctx)
+	return t
+}
+
+// TaskViewManagerImpl is the struct used to manage a single listView instance.
+// params
+// virtualCenter: holds a reference to the global VC object
+// sync.Mutex: is required to handle race conditions for
+// storing and retrieving tasks during a driver restart
+// taskToChannelMap: is an in-memory map to associate task details
+// to the channel created by the caller to receive the task result
+// context.Context: is separate from the context set by CSI ops
+// listView: holds the managed object used to monitor multiple concurrent VC tasks
+// shouldStopListening: in case of regular CSI operation, even after receiving a batch of updates,
+// we want to continue listening for subsequent updates.
+// in case of unit tests, we need to stop listening for the test to complete execution
+type TaskViewManagerImpl struct {
+	mu                  sync.Mutex
+	virtualCenter       *cnsvsphere.VirtualCenter
+	listView            *view.ListView
+	ctx                 context.Context
+	taskToChannelMap    map[TaskDetails]chan Result
+	shouldStopListening bool
+}
+
+func (t *TaskViewManagerImpl) ResetVirtualCenter(virtualCenter *cnsvsphere.VirtualCenter) {
+	t.mu.Lock()
+	defer t.mu.Unlock()
+	t.virtualCenter = virtualCenter
+}
+
+func getListViewWaitFilter(listView *view.ListView) *property.WaitFilter {
+	ts := types.TraversalSpec{
+		Type: "ListView",
+		Path: "view",
+		Skip: types.NewBool(false),
+	}
+	filter := new(property.WaitFilter)
+
+	filter.Add(listView.Reference(), "Task", []string{infoPropertyName}, &ts)
+	reportMissingObjectsInResults := true
+	filter.Spec.ReportMissingObjectsInResults = &reportMissingObjectsInResults
+	return filter
+}
+
+// listenToTaskUpdates is a long-running goroutine
+// that uses a property collector to listen for task updates
+// CSI ops add CNS tasks to listview and wait for a response from CNS
+// when update(s) are received by the property collector,
+// it spawns a new goroutine to process each task update and return the result to the caller
+func (t *TaskViewManagerImpl) listenToTaskUpdates(ctx context.Context) {
+	log := logger.GetLogger(ctx)
+	var filter *property.WaitFilter
+	recreateView := false
+	for {
+
+		if t.ctx.Err() == context.Canceled {
+			log.Info("listener canceled")
+			break
+		}
+
+		if t.getListView() == nil || recreateView {
+			log.Infof("listenToTaskUpdates -> recreateView -> createTasksListView")
```

Comment:
Please change the log message to something meaningful

Reply:
Fixed.